### DISCUSSION TITLE
rome format failing in ci

### DISCUSSION THREAD

**Discussion by igortas** [rome/tools#4299] [Q&A]
Hi,

I'm getting this issue when try to rune the format command in CI:

```
"format": "rome format src --write"
```

Dockerfile
```
RUN npm run format
```

Error
```
#0 0.993 /app/node_modules/rome/bin/rome:28
#0 0.993                throw result.error;
#0 0.993                ^
#0 0.993 
#0 0.993 Error: spawnSync /app/node_modules/@rometools/cli-linux-x64/rome ENOENT
#0 0.993     at Object.spawnSync (node:internal/child_process:1117:20)
#0 0.993     at Object.spawnSync (node:child_process:871:24)
#0 0.993     at Object.<anonymous> (/app/node_modules/rome/bin/rome:21:42)
#0 0.993     at Module._compile (node:internal/modules/cjs/loader:1275:14)
#0 0.993     at Module._extensions..js (node:internal/modules/cjs/loader:1329:10)
#0 0.993     at Module.load (node:internal/modules/cjs/loader:1133:32)
#0 0.993     at Module._load (node:internal/modules/cjs/loader:972:12)
#0 0.993     at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:83:12)
#0 0.993     at node:internal/main/run_main_module:23:47 {
#0 0.993   errno: -2,
#0 0.993   code: 'ENOENT',
#0 0.993   syscall: 'spawnSync /app/node_modules/@rometools/cli-linux-x64/rome',
#0 0.993   path: '/app/node_modules/@rometools/cli-linux-x64/rome',
#0 0.993   spawnargs: [ 'format', 'src', '--write' ]
#0 0.993 }
#0 0.993 
#0 0.993 Node.js v19.7.0
------
```

Rome: 11.0.0
Node.js: 19.7.0

**Comment by ematipico** [comment]
Rome library uses a post install script to install the binary for the correct platform, and it seems that the script didn't run.

You should make sure that your package manager runs the post install script inside docker.

**Comment by igortas** [comment]
@ematipico  Can u point more about that script for post install? Not sure what u mean... npm i -g rome@latest didn't help too.

**Reply by ematipico** [reply]
Rome runs a [`postinstall` script](https://docs.npmjs.com/cli/v9/using-npm/scripts?v=true#examples) to make sure that `rome` installed its dependencies, which doesn't seem the case. The package manager didn't install the [dependencies](https://www.npmjs.com/package/rome?activeTab=dependencies).

As a workaround, I would suggest to **install manually** the binary for you platform. `@rometools/cli-{PLATFORM}-{ARCHITETURE}`. For your particular case, you would need to install `@rometools/cli-linux-x64`

**Comment by igortas** [comment]
@ematipico No case works and spawn error still presented. Tried with:
- npm i -g
- /node_modules/path_to_rome
- installing binary with curl

and none of this help me when i hit the npm run format command.

How the docker should look?

```

COPY package*.json ./

RUN npm ci

RUN npm run format

COPY . .

RUN npm run build

```

**Reply by ematipico** [reply]
You have to add `@rometools/cli-linux-x64` to your package.json or install it manually via `npm install`.
